bin/p2-dsctl/main.go

```diff
@@ -163,11 +179,14 @@ func main() {
 		mutator := func(ds ds_fields.DaemonSet) (ds_fields.DaemonSet, error) {
 			changed := false
 
-			if *updateAZ != "" {
-				selector := selectorFrom(pc_fields.AvailabilityZone(*updateAZ))
+			if *updateSelector != SelectorNotSpecified {
```

Comment:
`if updateSelector != nil` -> then you know the selector was passed on the command line.